bluemira/equilibria/coils/_grouping.py

```diff
@@ -1015,3 +1136,118 @@ def from_group_vecs(
             for coil in self._get_coiltype(ctype)
         ]
         return self
+
+    def get_optimisation_state(
+        self, position_coil_names: list[str] | None = None, current_scale: float = 1.0
+    ) -> CoilSetOptimisationState:
+        """
+        Get the state of the CoilSet for optimisation
+
+        Parameters
+        ----------
+        position_coil_names:
+            The names of the coils to get the positions of
+        current_scale:
+            The scale of the currents
+
+        Returns
+        -------
+        CoilSetOptimisationState
+            The state of the CoilSet for optimisation
+        """
+        cc = self.get_control_coils()
+        xs, zs = cc._get_opt_positions(position_coil_names)
+        currents = cc.current / current_scale
+        return CoilSetOptimisationState(
+            currents=currents,
+            xs=xs,
+            zs=zs,
+        )
+
+    def set_optimisation_state(
+        self,
+        opt_currents: np.ndarray | None = None,
+        coil_position_map: dict[str, np.ndarray] | None = None,
+        current_scale: float = 1.0,
+    ):
+        """
+        Set the state of the CoilSet for optimisation
+
+        Parameters
+        ----------
+        opt_currents:
+            The optimisation currents
+        coil_position_map:
+            The map of coil names to positions
+        current_scale:
+            The scale of the currents
+        """
+        cc = self.get_control_coils()
+        if opt_currents is not None:
+            cc.current = opt_currents * current_scale
+        if coil_position_map is not None:
+            cc._set_opt_positions(coil_position_map)
+
+    @property
+    def n_position_optimisable_coils(self) -> int:
+        """
+        Get the number of coils that can be position optimised.
+        """
+        return len(self.position_optimisable_coil_names)
+
+    @property
+    def position_optimisable_coil_names(self) -> list[str]:
+        """
+        Get the names of the coils that can be position optimised.
+        """
+        optimisable_coil_names = [
+            c.primary_coil.name if isinstance(c, SymmetricCircuit) else c.name
+            for c in self._coils
+        ]
```

Comment:
I think if we change this to a generator we will only iterate through it once instead of twice
```suggestion
        optimisable_coil_names = (
            c.primary_coil.name if isinstance(c, SymmetricCircuit) else c.name
            for c in self._coils
        )
```